src/AdminApi/ScriptTag/ScriptTagService.php

```diff
@@ -0,0 +1,304 @@
+        $this->createNewScriptTagRequestBuilder = $createNewScriptTagRequestBuilder;
+        $this->deleteScriptTagRequestBuilder = $deleteScriptTagRequestBuilder;
+        $this->getScriptTagRequestBuilder = $getScriptTagRequestBuilder;
+        $this->getScriptTagsRequestBuilder = $getScriptTagsRequestBuilder;
+        $this->modifyExistingScriptTagRequestBuilder = $modifyExistingScriptTagRequestBuilder;
+        $this->pagedResultsIteratorBuilder = $pagedResultsIteratorBuilder;
+    }
+
+    /**
+     * Get a list of all script tags based on parameters set in the request.
+     *
+     * @see https://help.shopify.com/api/reference/scripttag#index
+     * @param GetScriptTagsRequest $request
+     * @return PagedResultsIterator|ScriptTag[]
+     */
+    public function getScriptTags(GetScriptTagsRequest $request): PagedResultsIterator
+    {
+        return $this->pagedResultsIteratorBuilder
+            ->withRequestBuilder($request)
+            ->withTransformer($this->scriptTagTransformer);
+    }
+
+    /**
+     * Async version of self::getScriptTags
+     *
+     * Please note that results will have to be manually transformed.
+     *
+     * @see https://help.shopify.com/api/reference/scripttag#index
+     * @param GetScriptTagsRequest $request
+     * @return PromiseInterface
+     */
+    public function asyncGetScriptTags(GetScriptTagsRequest $request): PromiseInterface
+    {
+        return $this->httpClient->sendAsync(
+            $request->toRequest(),
+            $request->toRequestOptions()
+        );
+    }
+
+    /**
+     * Count scripttags for a search criteria.
+     *
+     * @see https://help.shopify.com/api/reference/scripttag#count
+     * @param CountScriptTagsRequest $request
+     * @return int
+     * @throws MissingExpectedAttributeException
+     */
+    public function countScriptTags(CountScriptTagsRequest $request): int
+    {
+        $response = $this->asyncCountScriptTags($request)->wait();
+
+        $count = json_decode($response->getBody()->getContents());
+
+        if (!property_exists($count, 'count')) {
+            throw new MissingExpectedAttributeException('count');
+        }
+
+        return intval($count->count);
+    }
+
+    /**
+     * Async version of self::countScriptTags
+     *
+     * @see https://help.shopify.com/api/reference/scripttag#count
+     * @param CountScriptTagsRequest $request
+     * @return PromiseInterface
+     */
+    public function asyncCountScriptTags(CountScriptTagsRequest $request): PromiseInterface
+    {
+        return $this->httpClient->sendAsync(
+            $request->toRequest(),
+            $request->toRequestOptions()
+        );
+    }
+
+    /**
+     * Get an individual scripttag.
+     *
+     * @param RequestCredentialsInterface $credentials
+     * @param int $scriptTagId
+     * @param null|ScriptTagFields $scriptTagFields
+     * @return ScriptTag
+     */
+    public function getScriptTag(
```

Comment:
This should be renamed to `getScriptTagById`, I do plan to refactor existing endpoints to include this convention once I complete metafields.

If `getScriptTag` is to be kept, it should accept a `ScriptTag` model. Although more difficult for the caller, it's more consistent and reflective of the name.